rllib/agents/ppo/ddppo.py

```diff
@@ -129,6 +129,15 @@ def validate_config(self, config):
                 "Set rollout_fragment_length instead of train_batch_size "
                 "for DDPPO.")
 
+        # Check to make sure that the sgd_minibatch_size is not smaller
+        # than max_seq_len otherwise this will cause indexing errors while
+        # performing sgd when using a RNN or Attention model
+        if config["sgd_minibatch_size"] < config["model"]["max_seq_len"]:
```

Comment:
thinking about this a bit more, will this work for non-rnn stacks? there are a few tests failing because of this new check.
still feels a bit weird that sgd_minibatch_size is connected to max_seq_len now.
I wonder if this requirement should just be a warning or something.

Reply:
@gjoliver,

I think the question is what do you want to happen if a user sets max_seq_len > sgd_minibatch_size

The current code throws an exception but it only does so during training and only after the policy learns to generate episodes that last longer than sgd_minibatch_size. In StatelessCartpole it has to train for ~50 iterations before it asserts. That is a long time to wait to find out your config is set up incorrectly. The current assertion does not tell you that the config is the issue so you have to decode that.

One way to fix this is to slice based on min of (max_seq_len, sgd_minibatch_size) but if you do this you are ignoring the users request that truncated-bptt should be at most max_seq_len. So in the example in the issue you would be changing 200 -> =128. 


A second way to fix this is to test the configuration and warn the user ahead of time. The question is where to test. 

a. validate_config: Easy to test the config but no easy way to see if the model uses state and max_seq_len should even be considered. Currently some tests are failing because they fail the check but it should not matter because they do not use states.

b. At policy construction time. Then we could check policy.is_recurrent() and only if it is True do we do this config check.

c. Put it in do_minibatch_sgd. This might be the real place we want to put it since it is not actually about PPO but the sgd process.

Reply:
I just pushed an update that tries to implement option c.